### PR TITLE
Massive round of cleanup

### PR DESCRIPTION
**Pull request by mbabker**
- Fixes up the phpdoc task in build.xml to generate output for the API site
- Updates the coding-standards submodule
- Cleanup of various methods and doc blocks based on phpdoc run and phpStorm inspection

### CODE REVIEW

File: src/Joomla/Filesystem/Buffer.php [hunk 1]
```diff
@@ -45,18 +45,14 @@ class Buffer
...
-	 * @param   string   &$opened_path  Full path of the resource. Used with STREAN_USE_PATH option
+	 * @param   string   $path  The URL that was passed
 	 *
 	 * @return  boolean
 	 *
 	 * @since   1.0
 	 * @see     streamWrapper::stream_open
 	 */
-	public function stream_open($path, $mode, $options, &$opened_path)
+	public function stream_open($path)
```
Comment (mbabker) [inline src/Joomla/Filesystem/Buffer.php]: These were unused params, might as well get them out before we hit the v1 tag.

Comment (dongilbert) [inline src/Joomla/Filesystem/Buffer.php]: Are they required for being a stream_wrapper?

Comment (mbabker) [inline src/Joomla/Filesystem/Buffer.php]: Maybe?  I have no idea, but they're definitely not used here.

Comment (mbabker) [inline src/Joomla/Filesystem/Buffer.php]: I added them back for now.

**Pull request merged**